src/pages/ServiceDetails/ServiceDetailsPage.tsx

```diff
@@ -6,50 +6,77 @@ import ServiceId from '../../types/ServiceId';
 import { Nav, NavItem, TabContainer, TabContent, TabPane } from 'patternfly-react';
 import { NamespaceFilterSelected } from '../../components/NamespaceFilter/NamespaceFilter';
 import { ActiveFilter } from '../../types/NamespaceFilter';
+import * as API from '../../services/Api';
 
-const ServiceDetails = (routeProps: RouteComponentProps<ServiceId>) => {
-  let updateFilter = () => {
+type ServiceDetailsState = {
+  jaegerUri: string;
+};
+
+class ServiceDetails extends React.Component<RouteComponentProps<ServiceId>, ServiceDetailsState> {
+  constructor(props: RouteComponentProps<ServiceId>) {
+    super(props);
+    this.state = {
+      jaegerUri: ''
+    };
+  }
+  updateFilter = () => {
     let activeFilter: ActiveFilter = {
-      label: 'Namespace: ' + routeProps.match.params.namespace,
+      label: 'Namespace: ' + this.props.match.params.namespace,
       category: 'Namespace',
-      value: routeProps.match.params.namespace.toString()
+      value: this.props.match.params.namespace.toString()
     };
     NamespaceFilterSelected.setSelected([activeFilter]);
   };
-
-  return (
-    <div className="container-fluid container-pf-nav-pf-vertical">
-      <div className="page-header">
-        <h2>
-          Service{' '}
-          <Link to="/services" onClick={updateFilter}>
-            {routeProps.match.params.namespace}
-          </Link>{' '}
-          / {routeProps.match.params.service}
-        </h2>
-      </div>
-      <TabContainer id="basic-tabs" defaultActiveKey={1}>
-        <div>
-          <Nav bsClass="nav nav-tabs nav-tabs-pf">
-            <NavItem eventKey={1}>
-              <div dangerouslySetInnerHTML={{ __html: 'Info' }} />
-            </NavItem>
-            <NavItem eventKey={2}>
-              <div dangerouslySetInnerHTML={{ __html: 'Metrics' }} />
-            </NavItem>
-          </Nav>
-          <TabContent>
-            <TabPane eventKey={1}>
-              <ServiceInfo namespace={routeProps.match.params.namespace} service={routeProps.match.params.service} />
-            </TabPane>
-            <TabPane eventKey={2}>
-              <ServiceMetrics namespace={routeProps.match.params.namespace} service={routeProps.match.params.service} />
-            </TabPane>
-          </TabContent>
+  componentWillMount() {
+    API.getJaegerInfo().then(response => {
```

Comment:
Comment: I guess this assumes it never fails.
I guess for consistency it could be good to add the same error handling
(copy/paste what we have in other pages).

Reply:
I miss it... Added !!! @lucasponce ...we should in another PR show the errorMessage in App level...

Reply:
Yes, there is a global JIRA to unify the error handling as each page uses a complete different approach.